### ISSUE TITLE
How to manually create a `UnionSchema` schema?

### ISSUE THREAD

**Issue by bzm3r** [flavray/avro-rs#170]
I notice that the [`UnionSchema`](https://docs.rs/avro-rs/0.12.0/avro_rs/schema/struct.UnionSchema.html) is private, and thus cannot be accessed in order to create a union's schema directly. By that I mean: without passing a string, but instead creating a schema by composing `schema::Schema` types.

Is this on purpose?

**Comment by poros** [comment]
No, there isn't much of a reason for the `new` method to be `pub(crate)` while the other methods and the struct is actually public (the fields are not and I'd like to keep them that way, as one day we might want to fully support named types in unions and they will likely need to change). Would you like to do the change yourself?
